src/Controller/Controller.php

```diff
@@ -693,6 +697,22 @@ public function isAction($action)
     }
 
     /**
+     * {@inheritdoc}
+     *
+     * @param string $name Name of the event.
+     * @param array|null $data Any value you wish to be transported with this event to
+     * it can be read by listeners.
+     * @param \Cake\Controller\Controller|null $controller The controller that this event applies to
+     * ($this by default).
+     *
+     * @return \Cake\Controller\Event\ControllerEvent
+     */
+    public function dispatchEvent($name, $data = null, $controller = null)
+    {
+        return $this->_dispatchEvent($name, $data, $controller);
+    }
```

Comment:
Mainly to change the phpDoc as seen above the method. This way an IDE knows the method returns an ```ControllerEvent``` and expects as ```Controller``` as subject.